app/models/organization.rb

```diff
@@ -80,7 +81,12 @@ def update_from_json!(json)
       organization_json = parse json
 
       organization = Organization.find_by! name: organization_json[:name]
-      organization.update! organization_json
+      if organization_json[:book_ids] == organization.book_ids && organization_json[:book_id] == organization.book_id
```

Comment:
I hate this, but I need to check the condition before updating, and trigger reindexing afterwards. Maybe there's a smarter way to do this?

Reply:
Perhaps something like...

```ruby
old_book = organization.slice :book_id, :book_ids
...update as always...
reindex if old_book != organization_json.slice :book, :book_ids
```